factory/builder.py

```diff
@@ -207,18 +208,25 @@ def __init__(self, builder, sequence, parent_step=None):
         self.builder = builder
         self.sequence = sequence
         self.attributes = {}
+        self.deferred = {}
         self.parent_step = parent_step
         self.stub = None
 
-    def resolve(self, declarations):
+    def resolve(self, decls):
         self.stub = Resolver(
-            declarations=declarations,
+            declarations=decls,
             step=self,
             sequence=self.sequence,
         )
 
-        for field_name in declarations:
-            self.attributes[field_name] = getattr(self.stub, field_name)
+        for field_name in decls:
+            resolved = getattr(self.stub, field_name)
+
+            # `Maybe`s may include a post declaration, so defer them to the post build step
+            collector = self.deferred \
+                if isinstance(resolved, declarations.PostGenerationDeclaration) \
```

Comment:
This code removes any `PostGenerationDeclaration`s from the list of attributes used to create the object and puts them in a `deferred` map instead, which is used in `StepBuilder.build()` to move those resolved declarations to the post declarations.